### PR TITLE
fix: deploy.py

### PR DESCRIPTION
**Pull request by BobTheBuidler**

### CODE REVIEW

File: scripts/deploy.py [hunk 1]
```diff
@@ -1,58 +1,16 @@
...
 from eth_utils import is_checksum_address
-import click
-
-API_VERSION = config["dependencies"][0].split("@")[-1]
-Vault = project.load(
-    Path.home() / ".brownie" / "packages" / config["dependencies"][0]
-).Vault
-
-
-def get_address(msg: str, default: str = None) -> str:
```
Comment (fubuloubu) [inline scripts/deploy.py:13]: Keep this function

File: scripts/deploy.py [hunk 2]
```diff
@@ -1,58 +1,16 @@
...
-        # NOTE: Only display default once
-        val = click.prompt(msg)
 
+# For later use
+import click
 
 def main():
-    print(f"You are using the '{network.show_active()}' network")
-    dev = accounts.load(click.prompt("Account", type=click.Choice(accounts.load())))
-    print(f"You are using: 'dev' [{dev.address}]")
```
Comment (fubuloubu) [inline scripts/deploy.py:35]: Keep these lines

File: scripts/deploy.py [hunk 3]
```diff
@@ -1,58 +1,16 @@
...
-      name: '{vault.name()}'
-    symbol: '{vault.symbol()}'
-    """
-    )
-    publish_source = click.confirm("Verify source on etherscan?")
-    if input("Deploy Strategy? y/[N]: ").lower() != "y":
-        return
-
-    strategy = Strategy.deploy(vault, {"from": dev}, publish_source=publish_source)
+    token = "xxxxx"
```
Comment (fubuloubu) [inline scripts/deploy.py:11]:
```suggestion
    token = Contract.from_explorer(get_address("Underlying token"))
```

File: scripts/deploy.py [hunk 4]
```diff
@@ -1,58 +1,16 @@
...
+    registry = "0x3ee41c098f9666ed2ea246f4d2558010e59d63a0"
```
Comment (fubuloubu) [inline scripts/deploy.py:12]:
```suggestion
    registry = get_address("Registry address,", "v2.registry.ychad.eth")
```

File: scripts/deploy.py [hunk 5]
```diff
@@ -1,58 +1,16 @@
...
+    tokenName = "abcxyz"
+    tokenSymbol = "xxx"
```
Comment (fubuloubu) [inline scripts/deploy.py:14]: Get these from `token`

File: scripts/deploy.py [hunk 6]
```diff
@@ -1,58 +1,16 @@
...
+
```
Comment (fubuloubu) [inline scripts/deploy.py:15]: Add a verification of deployment step

File: scripts/deploy.py [hunk 7]
```diff
@@ -1,58 +1,16 @@
...
-    if input("Deploy Strategy? y/[N]: ").lower() != "y":
-        return
-
-    strategy = Strategy.deploy(vault, {"from": dev}, publish_source=publish_source)
+    token = "xxxxx"
+    registry = "0x3ee41c098f9666ed2ea246f4d2558010e59d63a0"
+    tokenName = "abcxyz"
+    tokenSymbol = "xxx"
+    
+    affiliatetoken = AffiliateToken.deploy(token, registry, tokenName, tokenSymbol, {"from": accounts[0]})
```
Comment (fubuloubu) [inline scripts/deploy.py:16]:
```suggestion
    affiliatetoken = AffiliateToken.deploy(token, registry, tokenName, tokenSymbol, {"from": dev})
    print(f"Deployed {affiliatetoken}")
```

**Review by fubuloubu — CHANGES_REQUESTED** [review]

File: scripts/deploy.py [hunk 8]
```diff
@@ -1,58 +1,17 @@
...
-        f"""
-    Strategy Parameters
-
-       api: {API_VERSION}
-     token: {vault.token()}
-      name: '{vault.name()}'
-    symbol: '{vault.symbol()}'
-    """
-    )
-    publish_source = click.confirm("Verify source on etherscan?")
```
Comment (fubuloubu) [inline scripts/deploy.py:54]: Example verification of deployment step

File: scripts/deploy.py [hunk 9]
```diff
@@ -1,58 +1,45 @@
...
-    if input("Deploy Strategy? y/[N]: ").lower() != "y":
-        return
-
-    strategy = Strategy.deploy(vault, {"from": dev}, publish_source=publish_source)
+    token = Contract.from_explorer(get_address("Underlying token"))
+    registry = get_address("Registry address,", "v2.registry.ychad.eth")
+    tokenName = token.name()
+    tokenSymbol = token.symbol()
+
```
Comment (fubuloubu) [inline scripts/deploy.py:41]:
```suggestion

    print(
        f"""
    Affiliate Token Parameters
     token: {token.address}
      name: '{tokenName}'
    symbol: '{tokenSymbol}'
    """
    )
    publish_source = click.confirm("Verify source on etherscan?")
    if input("Deploy Strategy? y/[N]: ").lower() != "y":
        return

```

File: scripts/deploy.py [hunk 10]
```diff
@@ -1,58 +1,45 @@
...
-    )
-    publish_source = click.confirm("Verify source on etherscan?")
-    if input("Deploy Strategy? y/[N]: ").lower() != "y":
-        return
-
-    strategy = Strategy.deploy(vault, {"from": dev}, publish_source=publish_source)
+    token = Contract.from_explorer(get_address("Underlying token"))
+    registry = get_address("Registry address,", "v2.registry.ychad.eth")
+    tokenName = token.name()
+    tokenSymbol = token.symbol()
```
Comment (fubuloubu) [inline scripts/deploy.py:40]: Both of these should be overriden somehow by the user

**Review by fubuloubu — CHANGES_REQUESTED** [review]

File: scripts/deploy.py [hunk 11]
```diff
@@ -1,58 +1,45 @@
 from pathlib import Path
 
-from brownie import Strategy, accounts, config, network, project, web3
+from brownie import AffiliateToken, Contract, accounts, config, network, project, web3
 from eth_utils import is_checksum_address
+
+# For later use
 import click
 
 API_VERSION = config["dependencies"][0].split("@")[-1]
```
[the remark this answers is not in the document]
Comment (BobTheBuidler) [reply scripts/deploy.py:9]: is this for the comment on line 6?

File: scripts/deploy.py [hunk 12]
```diff
@@ -28,31 +33,26 @@ def get_address(msg: str, default: str = None) -> str:
...
-       api: {API_VERSION}
-     token: {vault.token()}
-      name: '{vault.name()}'
-    symbol: '{vault.symbol()}'
+    Affiliate Token Parameters
+      api: {API_VERSION}
+      affiliate token name: {affiliateTokenName}
+      affiliate token symbol: {affiliateTokenSymbol}
+      vault: {token.name()}
+      symbol: '{token.symbol()}'
```
Comment (fubuloubu) [inline scripts/deploy.py:50]:
```suggestion
        name: '{affiliateTokenName}'
      symbol: '{affiliateTokenSymbol}'
       token: {token.address}
```

File: scripts/deploy.py [hunk 13]
```diff
@@ -28,31 +33,26 @@ def get_address(msg: str, default: str = None) -> str:
...
     """
     )
+    
     publish_source = click.confirm("Verify source on etherscan?")
-    if input("Deploy Strategy? y/[N]: ").lower() != "y":
-        return
```
Comment (fubuloubu) [inline scripts/deploy.py:56]: Keep this line, so user can abort if they don't like the deployment

File: scripts/deploy.py [hunk 14]
```diff
@@ -1,16 +1,21 @@
...
 
-from brownie import Strategy, accounts, config, network, project, web3
+from brownie import AffiliateToken, Contract, accounts, config, network, project, web3
 from eth_utils import is_checksum_address
+
 import click
 
+# User Input 
+affiliateTokenName = "abcdef"
+affiliateTokenSymbol = "ABC"
```
Comment (fubuloubu) [inline scripts/deploy.py:10]: User should set this from `click.input`, but I am okay with this I guess

**Review by fubuloubu — CHANGES_REQUESTED** [review]
Lint fixes and merge all commits into one

File: scripts/deploy.py (hunk 14 above)
Comment (BobTheBuidler) [reply scripts/deploy.py]: Fixed it anyway

File: scripts/deploy.py [hunk 15]
```diff
@@ -28,31 +29,30 @@ def get_address(msg: str, default: str = None) -> str:
...
+    Affiliate Token Parameters
+        name: '{affiliateTokenName}'
+      symbol: '{affiliateTokenSymbol}'
+       token: {token.address}
     """
     )
+    
     publish_source = click.confirm("Verify source on etherscan?")
+    
     if input("Deploy Strategy? y/[N]: ").lower() != "y":
```
Comment (fubuloubu) [inline scripts/deploy.py:53]:
```suggestion
    if input("Deploy Affiliate Token? y/[N]: ").lower() != "y":
```

**Review by fubuloubu — APPROVED** [review]
One minor change and good to go!

**Pull request closed without merging**